occurrences.go

```diff
@@ -29,3 +29,11 @@ func (r *occurrences) add(h string) bool {
 
 	return false
 }
+
+func newOccurrences(i time.Duration, c int) *occurrences {
+	return &occurrences{
+		registry: make(map[string][]time.Time),
```

Comment:
Is it ensured this will never be called concurrent? In no circumstances?

If not, a mutex is needed

Reply:
Not as part of the main logic. Don't try to make run the tests in parallel...